python/ml4ir/base/model/optimizers/optimizer.py

```diff
@@ -2,6 +2,19 @@
 from tensorflow.keras.optimizers.schedules import ExponentialDecay
 from ml4ir.base.model.optimizers import cyclic_learning_rate
 from ml4ir.base.config.keys import OptimizerKey, LearningRateScheduleKey, CyclicLearningRateType
+import tensorflow as tf
+
+class DefaultValues(object):
```

Comment:
Your class-string (aka docstring) is very good! Consider propagating part of the info in the class name e.g., `OptimizerDefaultValues`

Reply:
Came here to say this.

@mohazahran Think of it like this - you might have users doing `from optimizer import *`, which would import `DefaultValues` into the python context. Since `DefaultValues` is a very generic name, the user will have no idea what it really is. So as a rule of thumb, it is good to be as explicit as possible in class and function names. I am working on improving on this as well.